src/ui/templates/search/autocomplete.hbs

```diff
@@ -29,6 +29,7 @@
               data-eventtype="AUTO_COMPLETE_SELECTION"
               data-eventoptions='{"suggestedSearchText": "{{value}}"}'
               role="option"
+              id="yxt-AutoComplete-option-{{../../_config.name}}-{{@../index}}-{{@index}}"
```

Comment:
The `AutoComplete` component is not something people ever add directly. Do we provide a hook in the `SearchBar` for people to name the child `AutoComplete`? If no, I don't think there's really any way someone can set `_config.name` to have spaces in it.

If there is no mechanism to change the name, since `duplicateNamesAllowed` is false, then it shouldn't be possible to add more than one `AutoComplete`.

Reply:
What the searchcomponent does is set the autocomplete's component name to `${this.name}.autocomplete`, so there is kind of a mechanism to change the name you just have to also change your searchbar name

Reply:
the filtersearch component does the same thing when it adds an autocomplete